astroid/inference_tip.py

```diff
@@ -20,7 +21,11 @@
 
 _P = ParamSpec("_P")
 
-_cache: dict[tuple[InferFn, NodeNG], list[InferenceResult] | None] = {}
+_cache: dict[
+    tuple[InferFn, NodeNG, InferenceContext | None], list[InferenceResult] | None
```

Comment:
Typing improvement PR to come shortly, but can go ahead with this now, as it changes in this PR.

```suggestion
    tuple[InferFn, NodeNG, InferenceContext | None], list[InferenceResult]
```